docs/current/guides/899944-aws-cdk-ecs.md

```diff
@@ -0,0 +1,141 @@
+- Build, test and deploy an application to the AWS ECS cluster
+- Run all of the above through a Dagger pipeline
+
+:::tip
+The concepts demonstrated in this tutorial can be applied to any other Infrastructure as Code (IaC) tool. The code example shown below can also be reused to provision another infrastructure stack (such as Amazon EKS, AWS Lambda and others). Reusing the code example for your own needs is covered in Appendix A.
+:::
+
+## Requirements
+
+This tutorial assumes that:
+
+- You have a basic understanding of the [Go programming language](https://go.dev/).
+- You have a basic understanding of [AWS CloudFormation](https://aws.amazon.com/cloudformation/getting-started/).
+- You have access to a [AWS IAM user with permissions to create new resources on an AWS region](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_users_create.html#id_users_create_console).
+- Your [AWS IAM user has access keys](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html#Using_CreateAccessKey) and those keys are [configured to be used from your local host](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html).
+- You have [installed the AWS CDK CLI](https://docs.aws.amazon.com/cdk/v2/guide/cli.html) on your local host (only required for bootstrapping the CDK for a specific region).
+
+## Step 1: Bootstrap the AWS CDK
+
+The AWS CDK stores its state in an AWS CloudFormation stack named `CDKToolkit`. This stack needs to be present on every AWS region where you are managing resources using the AWS CDK.
+
+In order to bootstrap the AWS CDK for a region, simply run the following command from a terminal. replace the `AWS-ACCOUNT-NUMBER` and `AWS-REGION` placeholders with the corresponding AWS account details.
+
+```shell
+cdk bootstrap AWS-ACCOUNT-NUMBER/AWS-REGION
+```
+
+More information regarding this step is available in the [AWS CDK CLI documentation](https://docs.aws.amazon.com/cdk/v2/guide/cli.html#cli-bootstrap).
+
+## Step 2: Create the Dagger pipeline
+
+The example application used in this tutorial is a simple React application. Once the AWS CDK is bootstrapped, the next step is to create a Dagger pipeline to build, publish and deploy this example application.
+
+Obtain the Dagger pipeline code and its related helper functions from GitHub, as below:
+
+```shell
+git clone https://github.com/dagger/examples.git
+cd ./go/aws-cdk
+```
+
+This code is organized as follows:
+
+- `main.go`: This file contains the Dagger pipeline that builds the application, builds the container image of the application, publishes it and calls the AWS CDK to interface with the AWS infrastructure.
+- `aws.go`: This file contains helper functions for use with the AWS CDK CLI and the AWS API.
+- `registry.go`: This file contains helper functions to initialize the AWS ECR registry.
```

Comment:
@samalba I added this line as the previous text did not mention this file although it is part of the repo. However it made me wonder: why are the helpder functions split across two files (aws.go and registry.go)?